lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -0,0 +1,224 @@
+            token,
+            tokenType: validationParams.validTypes[0]
+        };
+    }
+    
+    /**
+     * Function to return JWKS (JSON Web Key Set) from parameters provided, jwks_uri provided, or from well-known endpoint
+     * @param {@link (BaseValidationParameters:type)}
+     * @returns 
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async getJWKS(validationParams: BaseValidationParameters): Promise<any> {
+        this.logger.trace("TokenValidator.getJWKS called");
+        
+        // Prioritize keystore or jwksUri if provided
+        if (validationParams.issuerSigningKeys) {
+            this.logger.verbose("TokenValidator - issuerSigningKeys provided");
+            return createLocalJWKSet({
+                keys: validationParams.issuerSigningKeys
+            });
+        } 
+        
+        if (validationParams.issuerSigningJwksUri) {
+            this.logger.verbose("TokenValidator - Creating JWKS from JwksUri");
+            return createRemoteJWKSet(new URL(validationParams.issuerSigningJwksUri));
+        }
+
+        // Do resiliency well-known endpoint thing here
+        const retrievedJwksUri: string = await this.openIdConfigProvider.fetchJwksUriFromEndpoint();
+        this.logger.verbose("TokenValidator - Creating JWKS from default");
+        return createRemoteJWKSet(new URL(retrievedJwksUri));
+    }
+
+    /**
+     * Function to check that validIssuers parameter is not an empty array. Throws emptyIssuerError if empty.
+     * @param {@link (BaseValidationParameters:type)}
+     * @returns 
+     */
+    setIssuerParams(options: BaseValidationParameters): string[] {
+        this.logger.trace("TokenValidator.setIssuerParams called");
+
+        // Check that validIssuers is not empty
+        if (options.validIssuers.length < 1 || options.validIssuers[0].length < 1) {
+            throw ValidationConfigurationError.createEmptyIssuerError();
+        }
+
+        return options.validIssuers;
+    }
+
+    /**
+     * Function to check that validAudiences parameter is not an empty array. Throws emptyAudience error if empty.
+     * @param {@link (BaseValidationParameters:type)} 
+     * @returns 
+     */
+    setAudienceParams(options: BaseValidationParameters): string[] {
+        this.logger.trace("TokenValidator.setAudienceParams called");
+
+        // Check that validAudiences is not empty
+        if (options.validAudiences.length < 1 || options.validAudiences[0].length < 1) {
+            throw ValidationConfigurationError.createEmptyAudienceError();
+        }
+
+        return options.validAudiences;
+    }
+ 
+    /**
+     * Function to validate additional claims on token, including nonce, c_hash, and at_hash. No return, throws error if invalid.
+     * @param payload JWT payload
+     * @param {@link (BaseValidationParameters:type)} 
+     */
+    async validateClaims(payload: JWTPayload, validationParams: BaseValidationParameters): Promise<void> {
```

Comment:
Should we also document the specific claims being validated here in comments? or is this covered in `BaseValidationParameters` definition?

Reply:
Updated comments to be more clear about specific claims. Additional claims may be added as public APIs are added.